share/pot/start.sh

```diff
@@ -591,28 +587,58 @@ _js_start()
 	sleep 0.5
 	pkill -f -j "$_pname" "^sleep 1234$"
 
+	# Here is where the pot is makred as started
+	lockf "${POT_TMP:-/tmp}/pot-lock-$_pname" "${_POT_PATHNAME}" set-status -p "$_pname" -s started
+	rc=$?
+	if [ $rc -eq 2 ]; then
+		_info "pot $_pname is already started (???)"
+	fi
+	if [ $rc -eq 1 ]; then
+		# should we retry (in case it's stopping?)
+		_error "pot $_pname is not in a state where it can be started"
+		# not returning, it could be catastrophic, but the situation is quite messed up
+	fi
+
 	wait "$_wait_pid"
 	_exit_code=$?
 
 	echo "{ \"ExitCode\": $_exit_code }" > "$_confdir/.last_run_stats"
 
 	if [ "$_persist" = "NO" ]; then
 		# non-persistent jails always need to die
-		if [ ! -e "${POT_TMP:-/tmp}/pot_stopped_${_pname}" ]; then
-			start-cleanup "$_pname" "${_iface}"
+		# Here is where the pot is stopping
+		lockf "${POT_TMP:-/tmp}/pot-lock-$_pname" "${_POT_PATHNAME}" set-status -p "$_pname" -s stopping
+		rc=$?
+		if [ $rc -eq 2 ]; then
+			_debug "pot $_pname is already stopping (maybe by a pot stop)"
+			return 0
 		fi
-		rm -f "${POT_TMP:-/tmp}/pot_stopped_${_pname}"
-
+		if [ $rc -eq 1 ]; then
+			# should we retry (in case it's stopping?)
+			_error "pot $_pname is not in a state where it can be stopped"
+			# returning, but the situation is quite messed up
```

Comment:
Again, this would probably mean that it is currently starting?

Reply:
yes, it's the only case for a -1
I'm assuming that if it's starting (from another invocation) no clean up is needed anymore because the new "starting" has already fixed things.
I cannot really imagine how this can happen, but I want to log it just in case